src/components/stateful-enroll/data/service.js

```diff
@@ -1,59 +1,47 @@
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
 import { camelCaseObject } from '@edx/frontend-platform/utils';
 import { getConfig } from '@edx/frontend-platform/config';
-import { v4 as uuidv4 } from 'uuid';
 
 /**
  * Makes an API request to retrieve the most recent payload for the
  * specified transaction UUID.
  *
- * Note: this arg may change to be given the actual API url to call
- * rather than just the `transactionUUID` (TBD during API integration).
- *
  * @param {object} args
- * @param {string} args.contentKey
  * @param {string} args.transactionUUID
  * @returns The payload for the specified transaction.
  */
 /* istanbul ignore next */
-export const retrieveTransactionStatus = async ({ contentKey, transactionUUID }) => {
-  const mockCommittedTransaction = {
-    uuid: transactionUUID,
-    state: 'committed',
-    courseware_url: `${getConfig().LEARNING_BASE_URL}/course/${contentKey}/home`,
-  };
-  const url = 'https://httpbin.org/post';
-  const response = await fetch(url, {
-    method: 'POST',
-    headers: {
-      Accept: 'application/json',
-      'Content-Type': 'application/json',
-    },
-    body: JSON.stringify(mockCommittedTransaction),
-  });
-  const result = await response.json();
-  return camelCaseObject(result.json);
+export const retrieveTransactionStatus = async ({ transactionUUID }) => {
+  const config = getConfig();
+
+  const url = `${config.ENTERPRISE_SUBSIDY_BASE_URL}/api/v1/transactions/${transactionUUID}/`;
+  const response = await getAuthenticatedHttpClient()
+    .get(url)
+    .catch(error => { throw error; });
+
+  return camelCaseObject(response.data);
 };
 
 /**
  * Makes an API request to submit a redemption request for the specified
  * user and course run key.
  *
+ * @param {object} args
+ * @param {string} args.policyRedemptionUrl
+ * @param {string} args.userId
+ * @param {string} args.contentKey
  * @returns Payload from the redemption request.
  */
 /* istanbul ignore next */
-export const submitRedemptionRequest = async () => {
-  const mockTransactionResponse = {
-    uuid: uuidv4(),
+export const submitRedemptionRequest = async ({ policyRedemptionUrl, userId, contentKey }) => {
+  const requestBody = {
+    lms_user_id: userId,
+    content_key: contentKey,
   };
-  const url = 'https://httpbin.org/post';
-  const response = await fetch(url, {
-    method: 'POST',
-    headers: {
-      Accept: 'application/json',
-      'Content-Type': 'application/json',
-    },
-    body: JSON.stringify(mockTransactionResponse),
-  });
-  const result = await response.json();
-  return camelCaseObject(result.json);
+
+  const { data } = await getAuthenticatedHttpClient()
+    .post(policyRedemptionUrl, requestBody)
+    .catch(error => { throw error; });
```

Comment:
The `.catch` here likely isn't needed; if the API throws an exception, it'll already throw the error up to `useQuery` to handle.